### PR TITLE
Maven support, take two

### PR DESCRIPTION
**Pull request by fthomas**
This contains @stanislav-chetvertkov's code from #1193 for supporting Maven. I accidentally closed #1193 by merging #1429 that contained all commits of #1193. Therefore this PR is basically a revert of 584b500e4a6fe2aac5d97709bcd576b9c6b1a8b7 and a few changes to bring the Maven support in sync with current master.

### CODE REVIEW

File: modules/core/src/main/scala/org/scalasteward/core/buildsystem/maven/MavenAlg.scala [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+  ): BuildSystemAlg[F] = new BuildSystemAlg[F] {
+
+    import MavenParser._
+
+    override def containsBuild(repo: Repo): F[Boolean] = ???
+
+    override def getDependencies(repo: Repo): F[List[Scope.Dependencies]] =
+      for {
+        repoDir <- workspaceAlg.repoDir(repo)
+        listDependenciesCommand = mvnCmd(command.Clean, command.mvnDepList)
```
Comment (fthomas) [inline modules/core/src/main/scala/org/scalasteward/core/buildsystem/maven/MavenAlg.scala:52]: @stanislav-chetvertkov I'm wondering why `clean` is called here? Is it required by `mvnDepList`?

**Comment by fthomas** [comment]
Here are the first two pull requests created by this branch:
* https://github.com/scala-steward-org/test-repo-2/pull/1
* https://github.com/scala-steward-org/test-repo-2/pull/2

**Comment by fthomas** [comment]
This PR is now ready for review. Thanks to @stanislav-chetvertkov's work in #1193, this didn't require any substantial changes from me.

This PR contains the following changes:
* Adds a new `MavenAlg` which implements `BuildSystemAlg` and hooks it into `BuildSystemDispatcher`. Scala Steward now checks if a `pom.xml` file exists in a repository and will invoke `mvn` to extract dependencies and Maven repositories via the `dependency:list` and `dependency:list-repositories` goals. If the top-level directory of a repo also contains a `build.sbt`, Scala Steward will use both, sbt and Maven, to extract dependencies.
* Allows Scala Steward to edit `pom.xml` files regardless of the build tool used in a repository.
* Maven repositories that require credentials for read access are currently not supported.
* Running Scalafix migrations in Maven projects is currently not supported.

From my POV this is ready to merged now. Supporting Scalafix migrations and credentials for Maven repositories can be added in subsequent PRs.

File: modules/core/src/main/scala/org/scalasteward/core/buildsystem/maven/MavenAlg.scala (hunk 1 above)
Comment (stanislav-chetvertkov) [reply modules/core/src/main/scala/org/scalasteward/core/buildsystem/maven/MavenAlg.scala]: I guess it was an extra precaution, probably could be removed

**Pull request merged**